src/chunk.c

```diff
@@ -3520,9 +3455,21 @@ ts_chunk_clear_status(Chunk *chunk, int32 status)
 						   status,
 						   chunk->fd.status)));
 	}
-	uint32 mstatus = ts_clear_flags_32(chunk->fd.status, status);
-	chunk->fd.status = mstatus;
-	return chunk_update_status(&chunk->fd);
+
+	FormData_chunk form;
+	ItemPointerData tid;
+	bool found = lock_chunk_tuple(chunk->fd.id, &tid, &form);
+	Assert(found);
+
+	/* applying the flags after locking the metadata tuple */
+	int32 old_status = form.status;
+	form.status = ts_clear_flags_32(form.status, status);
```

Comment:
The check for frozen (at line 3447) should happen here as well: i.e. after locking the tuple , check if frozen bit is set in old_status.

Might be good to add a debug_waitpoint to trigger this error via the isolation tests.

Reply:
Same for ts_chunk_add_status, ts_chunk_set_compressed_chunk and ts_chunk_clear_compressed_chunk